src/renderer/state/route/logics.ts

```diff
@@ -164,98 +160,90 @@ function convertUploadRequestsToUploadStateBranch(
-          return keyToMetadataSoFar;
-        }
+        (
+            keyToMetadataSoFar: UploadStateBranch,
+            annotation: MMSFileAnnotation
+        ) => {
+          const annotationDefinition =
+              annotationIdToAnnotationMap[annotation.annotationId]?.[0];
+          // Only include annotations that are on the template in the metadata for the branch,
+          // otherwise this might try to overwrite metadata from other applications unintentionally
+          // But, do include Well and Notes, because they are included in the grid by default.
+          if (!setOfAnnotationIdsFromTemplate.has(annotation.annotationId) && annotationDefinition?.name !== AnnotationName.WELL && annotationDefinition?.name !== AnnotationName.NOTES) {
+            return keyToMetadataSoFar;
+          }
 
-        const key = getUploadRowKey({
-          file: file.file.originalPath,
-          ...annotation,
-        });
-        if (!annotationDefinition) {
-          throw new Error(
-            `Unable to find matching Annotation for Annotation ID: ${annotation.annotationId}`
-          );
-        }
+          if (!annotationDefinition) {
+            throw new Error(
+                `Unable to find matching Annotation for Annotation ID: ${annotation.annotationId}`
+            );
+          }
 
-        let values: any[] = annotation.values;
-        switch (annotationDefinition["annotationTypeId/Name"]) {
-          case ColumnType.BOOLEAN:
-            values = values.map((v) => `${v}`.toLowerCase() === "true");
-            break;
-          case ColumnType.DATE:
-          case ColumnType.DATETIME:
-            values = values.map((v) => new Date(`${v}`));
-            break;
-          case ColumnType.LOOKUP:
-            if (
-              annotationIdToLookupMap[annotation.annotationId]?.[
-                "scalarTypeId/Name"
-              ] === ScalarType.INT
-            ) {
-              values = values.map((v) => parseInt(v, 10));
-            }
-            break;
-          case ColumnType.NUMBER:
-            values = values.map((v) => {
-              try {
-                return parseFloat(v);
-              } catch (e) {
-                return v;
+          let values: any[] = annotation.values;
+          switch (annotationDefinition["annotationTypeId/Name"]) {
+            case ColumnType.BOOLEAN:
+              values = values.map((v) => `${v}`.toLowerCase() === "true");
+              break;
+            case ColumnType.DATE:
+            case ColumnType.DATETIME:
+              values = values.map((v) => new Date(`${v}`));
+              break;
+            case ColumnType.LOOKUP:
+              if (
+                  annotationIdToLookupMap[annotation.annotationId]?.[
+                      "scalarTypeId/Name"
+                      ] === ScalarType.INT
+              ) {
+                values = values.map((v) => parseInt(v, 10));
```

Comment:
The changes in this file look like a lot, but it's actually mostly whitespace. See [the diff ignoring whitespace here](https://gist.github.com/pgarrison/c915c847ddef2b331f5cb565ad190e82)